controllers/provider-aws/pkg/infrastructure/actuator_helpers.go

```diff
@@ -0,0 +1,267 @@
+	}
+
+	awsClient, err := awsclient.NewClient(string(providerSecret.Data[aws.AccessKeyID]), string(providerSecret.Data[aws.SecretAccessKey]), region)
+	if err != nil {
+		return nil, err
+	}
+	// check if we should use an existing VPC or create a new one
+	if infrastructureConfig.Networks.VPC.ID != nil {
+		createVPC = false
+		vpcID = *infrastructureConfig.Networks.VPC.ID
+		igwID, err := awsClient.GetInternetGateway(ctx, vpcID)
+		if err != nil {
+			return nil, err
+		}
+		internetGatewayID = igwID
+	} else if infrastructureConfig.Networks.VPC.CIDR != nil {
+		vpcCIDR = infrastructureConfig.Networks.VPC.CIDR
+	}
+
+	return map[string]interface{}{
+		"aws": map[string]interface{}{
+			"region": infrastructure.Spec.Region,
+		},
+		"create": map[string]interface{}{
+			"vpc": createVPC,
+		},
+		"sshPublicKey": string(infrastructure.Spec.SSHPublicKey),
+		"vpc": map[string]interface{}{
+			"id":                vpcID,
+			"cidr":              vpcCIDR,
+			"dhcpDomainName":    dhcpDomainName,
+			"internetGatewayID": internetGatewayID,
+		},
+		"clusterName": infrastructure.Namespace,
+		"zones":       getZones(infrastructureConfig),
+		"outputKeys": map[string]interface{}{
+			"vpcIdKey":                   aws.VPCIDKey,
+			"subnetsPublicPrefix":        aws.SubnetPublicPrefix,
+			"subnetsNodesPrefix":         aws.SubnetNodesPrefix,
+			"securityGroupsNodes":        aws.SecurityGroupsNodes,
+			"sshKeyName":                 aws.SSHKeyName,
+			"iamInstanceProfileNodes":    aws.IAMInstanceProfileNodes,
+			"iamInstanceProfileBastions": aws.IAMInstanceProfileBastions,
+			"nodesRole":                  aws.NodesRole,
+			"bastionsRole":               aws.BastionsRole,
+		},
+	}, nil
+}
+
+func getZones(infraProviderConfig *awsapi.InfrastructureConfig) []map[string]interface{} {
+	var zones []map[string]interface{}
+	for _, zone := range infraProviderConfig.Networks.Zones {
+		zones = append(zones, map[string]interface{}{
+			"name":     zone.Name,
+			"worker":   zone.Workers,
+			"public":   zone.Public,
+			"internal": zone.Internal,
+		})
+	}
+	return zones
+}
+
+func getSubnets(infrastructure *extensionsv1alpha1.Infrastructure, values map[string]string) ([]v1alpha1.Subnet, error) {
+	var subnetsToReturn []v1alpha1.Subnet
+	for key, value := range values {
+		if strings.HasPrefix(key, aws.SubnetPublicPrefix) {
+			zoneID, err := strconv.Atoi(strings.TrimPrefix(key, aws.SubnetPublicPrefix))
+			if err != nil {
+				return nil, err
+			}
+			subnetsToReturn = append(subnetsToReturn, v1alpha1.Subnet{
+				Name: key,
+				ID:   value,
+				Zone: infrastructure.Spec.Zones[zoneID],
+			})
+		}
+		if strings.HasPrefix(key, aws.SubnetNodesPrefix) {
+			zoneID, err := strconv.Atoi(strings.TrimPrefix(key, aws.SubnetNodesPrefix))
+			if err != nil {
+				return nil, err
+			}
+			subnetsToReturn = append(subnetsToReturn, v1alpha1.Subnet{
+				Name: key,
+				ID:   value,
+				Zone: infrastructure.Spec.Zones[zoneID],
+			})
```

Comment:
You have duplicated the code with `zoneID,err := ...` and `subnetsToReturn = ...`. Can you refactor to only have it once?